Remove non-standard `WheelEvent.initWebKitWheelEvent()`

Ahmad-S792 opened the pull request:
#### f50de885e127381197e0bab6e3a0f598161b2758
<pre>
Remove non-standard `WheelEvent.initWebKitWheelEvent()`

<a href="https://bugs.webkit.org/show_bug.cgi?id=267813">https://bugs.webkit.org/show_bug.cgi?id=267813</a>
<a href="https://rdar.apple.com/problem/121733957">rdar://problem/121733957</a>

Reviewed by Ryosuke Niwa.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium and Web Specification [1]:

[1] <a href="https://w3c.github.io/uievents/#interface-wheelevent">https://w3c.github.io/uievents/#interface-wheelevent</a>

This patch remoes prefixed non-standard API `initWebKitWheelEvent` from WheelEvent, it was
never supported by Gecko and removed by Blink in 2014.

All test changes are merged from below Blink commit:

Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=179230

* Source/WebCore/dom/WheelEvent.cpp:
(WheelEvent::initWebKitWheelEvent): Deleted
* Source/WebCore/dom/WheelEvent.h:
* Source/WebCore/dom/WheelEvent.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMWheelEvent.cpp:
(webkit_dom_wheel_event_init_wheel_event):
* Source/WebKitLegacy/mac/DOM/DOMWheelEvent.mm:
(initWheelEvent):
* LayoutTests/fast/events/wheelevent-in-scrolling-div.html: Rebaselined
* LayoutTests/fast/events/wheelevent-in-scrolling-div-expected.txt: Ditto
* LayoutTests/fast/events/wheelevent-delta-overflow.html: Ditto
* LayoutTests/fast/events/wheelevent-delta-overflow-expected.txt: Ditto
* LayoutTests/fast/forms/resources/common-wheel-event.js: Update Test Script

Canonical link: <a href="https://commits.webkit.org/274069@main">https://commits.webkit.org/274069@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f9b6161bb081aba5b98929d04826314245973a5b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37776 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16671 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40017 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40316 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33603 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/39103 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19319 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13896 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31959 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38343 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14052 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33065 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12261 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12190 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33773 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41575 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34178 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34195 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38081 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12786 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10355 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36256 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14198 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/33167 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8488 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13166 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13509 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
LGTM

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMWheelEvent.cpp:
```diff
@@ -130,8 +130,7 @@ void webkit_dom_wheel_event_init_wheel_event(WebKitDOMWheelEvent* self, glong wh
     g_return_if_fail(WEBKIT_DOM_IS_DOM_WINDOW(view));
-    WebCore::WheelEvent* item = WebKit::core(self);
-    item->initWebKitWheelEvent(wheelDeltaX, wheelDeltaY, WebKit::toWindowProxy(view), screenX, screenY, clientX, clientY, ctrlKey, altKey, shiftKey, metaKey);
+    return;
```
No need to write `return` at the end of a void function

Ahmad-S792 replied:
Done in [f9b6161](https://github.com/WebKit/WebKit/commit/f9b6161bb081aba5b98929d04826314245973a5b).

mcatanzaro commented on Source/WebKitLegacy/mac/DOM/DOMWheelEvent.mm:
```diff
@@ -95,8 +95,7 @@ - (BOOL)isHorizontal
 {
-    WebCore::JSMainThreadNullState state;
-    IMPL->initWebKitWheelEvent(inWheelDeltaX, inWheelDeltaY, toWindowProxy(view), screenX, screenY, clientX, clientY, ctrlKey, altKey, shiftKey, metaKey);
+    return;
```
Ditto

Ahmad-S792 replied:
Done in [f9b6161](https://github.com/WebKit/WebKit/commit/f9b6161bb081aba5b98929d04826314245973a5b).

mcatanzaro reviewed (commented):
The GTK change is fine (other than the style nit)

Ahmad-S792 committed:
Remove non-standard `WheelEvent.initWebKitWheelEvent()`

https://bugs.webkit.org/show_bug.cgi?id=267813
rdar://problem/121733957

Reviewed by Ryosuke Niwa.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium and Web Specification [1]:

[1] https://w3c.github.io/uievents/#interface-wheelevent

This patch remoes prefixed non-standard API `initWebKitWheelEvent` from WheelEvent, it was
never supported by Gecko and removed by Blink in 2014.

All test changes are merged from below Blink commit:

Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=179230

* Source/WebCore/dom/WheelEvent.cpp:
(WheelEvent::initWebKitWheelEvent): Deleted
* Source/WebCore/dom/WheelEvent.h:
* Source/WebCore/dom/WheelEvent.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMWheelEvent.cpp:
(webkit_dom_wheel_event_init_wheel_event):
* Source/WebKitLegacy/mac/DOM/DOMWheelEvent.mm:
(initWheelEvent):
* LayoutTests/fast/events/wheelevent-in-scrolling-div.html: Rebaselined
* LayoutTests/fast/events/wheelevent-in-scrolling-div-expected.txt: Ditto
* LayoutTests/fast/events/wheelevent-delta-overflow.html: Ditto
* LayoutTests/fast/events/wheelevent-delta-overflow-expected.txt: Ditto
* LayoutTests/fast/forms/resources/common-wheel-event.js: Update Test Script

Canonical link: https://commits.webkit.org/274069@main
